platform/include/transport_interface.h

```diff
@@ -19,21 +19,144 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
+/**
+ * @file transport_interface.h
+ * @brief Transport interface definitions to send and receive data over the
+ * network.
+ */
 #ifndef TRANSPORT_INTERFACE_H_
 #define TRANSPORT_INTERFACE_H_
 
 #include <stdint.h>
 #include <stddef.h>
 
 /**
+ * @page transport_interface Transport Interface
+ * @brief The transport interface implementation
+ *
+ * The transport interface is a set of APIs that must be implemented using an
+ * external transport layer protocol. The transport interface is implemented in
+ * @ref transport_interface.h. This interface allows the application layer
+ * protocol library to send and receive data over the transport layer. This
+ * interface does not handle connection and disconnection to the server of
+ * interest. The connection to the server of interest must be made before the
+ * library is used with @ref TransportInterface_t. The application is also
+ * responsible for disconnection from the server. This interface does not handle
+ * socket timeouts or any setup of TLS; these things must be done before using
+ * the library's functions that need to send and receive data.
+ * <br>
+ *
+ * The functions that must be implemented are:<br>
+ * - [Transport Receive](@ref TransportRecv_t)
+ * - [Transport Send](@ref TransportSend_t)
+ *
+ * Each of the functions above take in an opaque context @ref NetworkContext_t.
+ * The functions above and the context are also grouped together in the
+ * @ref TransportInterface_t structure:<br><br>
+ * @snippet this define_transportinterface
+ * <br>
+ *
+ * The following steps give guidance on implementing the transport interface:
+ *
+ * -# Implementing the @ref NetworkContext_t<br><br>
+ * @snippet this define_networkcontext
+ * <br>
+ * @ref NetworkContext_t is the incomplete type <b>struct NetworkContext</b>.
+ * The implemented struct NetworkContext should contain all of the information
+ * that is needed to receive and send data with the @ref TransportRecv_t
+ * and the @ref TransportSend_t implementations.<br>
+ * struct NetworkContext is typically implemented with the TCP socket context
```

Comment:
In case of TLS over TCP, struct NetworkContext is typically implemented with the TCP socket context and a TLS context.

Reply:
thanks